kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetDescriptor.java

```diff
@@ -935,6 +936,21 @@ public DatasetDescriptor build() {
         columnMapping = copiedMapping;
       }
 
+      // Parse for properties in the entity schema and merge them with
+      // properties defined. If a property is defined using this builder and
+      // is also in the schema, the value set in the builder will take precedence
+      if(EntityPropertiesParser.hasEmbeddedProperties(schema)) {
```

Comment:
This pattern was needed for the partition strategy and column mappings because we wanted to provide backward-compatibility for existing HBC schemas. Since properties are an addition to the SchemaManager, I don't think we need to worry about them being embedded in Schemas. I think that removes quite a bit of code!